api/user_profile.go

```diff
@@ -30,18 +35,53 @@ func (svc *UserProfileService) Get() (response UserProfileResponse, err error) {
 }
 
 type UserProfileResponse struct {
-	Data []struct {
-		Username   string `json:"username"`
-		OrgAccount bool   `json:"orgAccount"`
-		URL        string `json:"url"`
-		OrgAdmin   bool   `json:"orgAdmin"`
-		OrgUser    bool   `json:"orgUser"`
-		Accounts   []struct {
-			Admin       bool   `json:"admin"`
-			AccountName string `json:"accountName"`
-			CustGUID    string `json:"custGuid"`
-			UserGUID    string `json:"userGuid"`
-			UserEnabled int    `json:"userEnabled"`
-		} `json:"accounts"`
-	} `json:"data"`
+	Data []UserProfile `json:"data"`
+}
+
+type UserProfile struct {
+	Username   string    `json:"username"`
+	OrgAccount bool      `json:"orgAccount"`
+	URL        string    `json:"url"`
+	OrgAdmin   bool      `json:"orgAdmin"`
+	OrgUser    bool      `json:"orgUser"`
+	Accounts   []Account `json:"accounts"`
+}
+
+func (p *UserProfile) OrgAccountName() string {
+	// TODO @afiune should we handle other datacenters?
+	rx, err := regexp.Compile(`\.lacework\.net.*`)
+	if err == nil {
+		if split := rx.Split(p.URL, -1); len(split) != 0 {
+			return strings.ToLower(split[0])
+		}
+	}
+
+	return p.URL
+}
```

Comment:
Are there any tests for this regex? Ideally that tests US and EU.

Reply:
Now they are 🙌🏽

![tenor-67817893](https://user-images.githubusercontent.com/5712253/119903664-3f70d400-bf06-11eb-9729-8ff59c53e1e2.gif)